### PR TITLE
New module: web_graph_improved.

### PR DESCRIPTION
**Pull request by LeartS**
This is a module to improve the charting capabilities of Odoo, which are quite limited.
At the moment the only thing this module does is allow you to plot multiple measures when there is only a single row group, but I plan (and anybody can help) on adding a lot more functions.

Before:
screenshot from 2015-01-08 12 46 50
After:
screenshot from 2015-01-08 12 55 15
More than 2:
screenshot from 2015-01-08 12 55 43

### CODE REVIEW

File: web_graph_improved/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Charts, Improved.
+=================
+
+Various improvements to the graph view:
+
+ * Bar: can plot multiple variables (called measures in the odoo client)
+   when there is only one row group
```
Comment (gurneyalex) [inline web_graph_improved/README.md]: various improvement_s_, and then you list one. Is it an omission or are you planning more improvements in the future?

File: web_graph_improved/static/src/js/web_graph_improved.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+openerp.web_graph_improved = function(instance) {
```
Comment (gurneyalex) [inline web_graph_improved/static/src/js/web_graph_improved.js:1]: no license declaration here?

File: web_graph_improved/README.md (hunk 1 above)
Comment (gurneyalex) [inline web_graph_improved/README.md]: Saw the description in the PR initial comment. The description here should be improved a bit, IMO.

**Comment by gurneyalex** [comment]
:+1: once the README.md file is updated. 

A small test would be welcome.

File: web_graph_improved/static/src/js/web_graph_improved.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                            y: (subhdr) ? self.pivot.get_total(subhdr)[0] : 0
+                        };
+                    });
+                    return {key:key, values: values};
+                });
+            }
+
+            nv.addGraph(function () {
+                var chart = nv.models.multiBarChart()
+                // .reduceXTicks(false)
```
Comment (foutoucour) [inline web_graph_improved/static/src/js/web_graph_improved.js]: Remove not used lines, or comment to explain why they are not ran.

Thanks

File: web_graph_improved/static/src/js/web_graph_improved.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    .stacked(self.bar_ui === 'stack')
+                    .showControls(show_controls);
+
+                // if (self.width / data[0].values.length < 80) {
```
Comment (foutoucour) [inline web_graph_improved/static/src/js/web_graph_improved.js]: same

**Comment by foutoucour** [comment]
:+1: Once the not used line case is managed

**Comment by legalsylvain** [comment]
Hi @LeartS ,
Can you take the time to change your code, regarding @foutoucour's comment ?

Regards.

File: web_graph_improved/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+{
+    'name': 'Better Charts',
+    'version': '0.1',
+    'category': 'Web',
+    'summary': 'Improves graph views.',
+    'author': 'Leonardo Donelli',
+    'website': 'http://learts.glaucus.in',
+    'depends': ['web_graph'],
+    'data': ['view/graph_improved_view.xml'],
+    'installable': True,
```
Comment (legalsylvain) [inline web_graph_improved/__openerp__.py:32]: Please can you add AGPL licence in the dictionnary ?

**Comment by LeartS** [comment]
Removed commented lines, add license, rebased.

**Comment by dreispt** [comment]
It would be nice to include the screenshots you provided in the PR description. I believe they should go in a `description/` dir.

**Comment by hbrunn** [comment]
:+1: but you might want to add the screenshots as @dreispt proposes and the license as @gurneyalex proposes in a new PR

**Pull request merged**